### PR TITLE
[GNA] support for 3D [1,1,X] import/output

### PR DESCRIPTION
**Pull request by aalbersk**
Jira: 41676

### CODE REVIEW

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 1]
```diff
@@ -872,14 +872,18 @@ uint32_t GNAPlugin::QueueInference(const InferenceEngine::BlobMap &inputs, Infer
...
     for (auto &input : inputs) {
         auto inputLayout = input.second->getTensorDesc().getLayout();
-        if (inputLayout != Layout::NC && inputLayout != Layout::CN && inputLayout != NCHW) {
+        if (inputLayout != Layout::NC && inputLayout != Layout::CN && inputLayout != Layout::CHW && inputLayout != NCHW) {
             THROW_GNA_EXCEPTION << "Expected input blob to have Layout::NC or Layout::CN, but was: "
                                 << input.second->getTensorDesc().getLayout();
         }
-        if (inputLayout == NCHW) {
-            inputLayout = NC;
+        if (inputLayout == Layout::NCHW || inputLayout == Layout::CHW) {
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:879]: only if H == 1 and W == 1

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 2]
```diff
@@ -904,10 +914,8 @@ uint32_t GNAPlugin::QueueInference(const InferenceEngine::BlobMap &inputs, Infer
             }
         }
 
-        auto dims = input.second->getTensorDesc().getDims();
-
         auto  importedElements = is2D ? dims[dims.size() - 1] : dims[dims.size() - 1] * dims[dims.size() - 2] * dims[dims.size() - 3];
-        auto  importedFrames = dims[0];
+        auto  importedFrames = is3D ? 1 : dims[0];
         auto  targetGroups = is2D ? dims[dims.size() - 2] : dims[0]; // TODO: no proper support for groups yet
```
Comment (kmagiers) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:919]: not sure, should it not be dims[dims.size()-1] for 3d?

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 3]
```diff
@@ -1034,8 +1042,9 @@ GnaWaitStatus GNAPlugin::WaitFor(uint32_t request_idx, int64_t millisTimeout) {
 //                           dims[dims.size() - 1]);
 //        }
             auto is2D = outputBlob->getTensorDesc().getLayout() == Layout::NC || outputBlob->getTensorDesc().getLayout() == Layout::CN;
+            auto is3D = outputBlob->getTensorDesc().getLayout() == Layout::CHW;
             auto& exportOutputDims = outputBlob->getTensorDesc().getDims();
-            auto batchSize = exportOutputDims[0];
+            auto batchSize = is3D ? 1 : exportOutputDims[0];
             auto elementsPerBatch = is2D ? exportOutputDims[exportOutputDims.size() - 1]
```
Comment (kmagiers) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:1048]: 3D case?

File: inference-engine/src/gna_plugin/gna_plugin.cpp [hunk 4]
```diff
@@ -872,14 +872,24 @@ uint32_t GNAPlugin::QueueInference(const InferenceEngine::BlobMap &inputs, Infer
...
-        if (inputLayout != Layout::NC && inputLayout != Layout::CN && inputLayout != NCHW) {
+        if (inputLayout != Layout::NC && inputLayout != Layout::CN && inputLayout != Layout::CHW && inputLayout != Layout::NCHW) {
             THROW_GNA_EXCEPTION << "Expected input blob to have Layout::NC or Layout::CN, but was: "
                                 << input.second->getTensorDesc().getLayout();
         }
-        if (inputLayout == NCHW) {
-            inputLayout = NC;
+
+        auto dims = input.second->getTensorDesc().getDims();
+        if (inputLayout == Layout::CHW && (dims[0] != 0 || dims[1] != 0)) {
```
Comment (kmagiers) [inline inference-engine/src/gna_plugin/gna_plugin.cpp:881]: != 1 instead of 0

**Review by kmagiers — APPROVED** [review]

File: inference-engine/src/gna_plugin/gna_plugin.cpp (hunk 2 above)
Comment (aalbersk) [reply inference-engine/src/gna_plugin/gna_plugin.cpp:919]: 3D case is CHW - it is a layout for a single image

**Pull request merged**